src/modules/secrets/Elsa.Secrets/Manager/SecretsManager.cs

```diff
@@ -7,45 +8,180 @@
 using System.Threading.Tasks;
 using Elsa.Secrets.Persistence.Specifications;
 using System.Linq;
+using Elsa.Secrets.Encryption;
+using Microsoft.Extensions.Configuration;
 
 namespace Elsa.Secrets.Manager
 {
     public class SecretsManager : ISecretsManager
     {
         private readonly ISecretsStore _secretsStore;
-        public SecretsManager(ISecretsStore secretsStore)
-        { 
+        private readonly bool _encryptionEnabled;
+        private readonly string _encryptionKey;
+        private readonly string[] _encryptedProperties;
+
+        public SecretsManager(ISecretsStore secretsStore, IConfiguration configuration)
```

Comment:
Rather than directly injecting `IConfiguration`, I'd recommend using an options object. This approach not only grants greater flexibility to the program's Startup code but also aligns with the convention followed by other classes that inject options.